gpmp2/kinematics/Arm.h

```diff
@@ -25,80 +25,98 @@ namespace gpmp2 {
 
-  Arm(size_t dof, const gtsam::Vector& a, const gtsam::Vector& alpha,
-      const gtsam::Vector& d, const gtsam::Pose3& base_pose,
-      const gtsam::Vector& theta_bias);
+  Arm(size_t dof, const gtsam::Vector &a, const gtsam::Vector &alpha,
+      const gtsam::Vector &d, const gtsam::Pose3 &base_pose,
+      const gtsam::Vector &theta_bias, const bool &modDH = false);
 
   /// Default destructor
   virtual ~Arm() {}
 
   /**
    *  Forward kinematics: joint configuration to poses in workspace
    *  Velocity kinematics: optional joint velocities to linear velocities in
-   *workspace, no anuglar rate
+   *  workspace, no angular rate
    *
    *  @param jp joint position in config space
    *  @param jv joint velocity in config space
    *  @param jpx joint pose in work space
    *  @param jvx joint velocity in work space
    *  @param J_jpx_jp et al. optional Jacobians
    **/
-  void forwardKinematics(const gtsam::Vector& jp,
+  void forwardKinematics(const gtsam::Vector &jp,
                          std::optional<const gtsam::Vector> jv,
-                         std::vector<gtsam::Pose3>& jpx,
-                         std::vector<gtsam::Vector3>* jvx = nullptr,
+                         std::vector<gtsam::Pose3> &jpx,
+                         std::vector<gtsam::Vector3> *jvx = nullptr,
                          gtsam::OptionalMatrixVecType J_jpx_jp = nullptr,
                          gtsam::OptionalMatrixVecType J_jvx_jp = nullptr,
                          gtsam::OptionalMatrixVecType J_jvx_jv = nullptr) const;
 
   /// update base pose in const
-  void updateBasePose(const gtsam::Pose3& p) const { base_pose_ = p; }
-
-  /// accesses
-  const gtsam::Vector& a() const { return a_; }
-  const gtsam::Vector& d() const { return d_; }
-  const gtsam::Vector& alpha() const { return alpha_; }
-  const gtsam::Pose3& base_pose() const { return base_pose_; }
+  void updateBasePose(const gtsam::Pose3 &p) const { base_pose_ = p; }
+
+  /// accessors
+  const gtsam::Vector &a() const { return a_; }
+  const gtsam::Vector &d() const { return d_; }
+  const gtsam::Vector &alpha() const { return alpha_; }
+  const gtsam::Pose3 &base_pose() const { return base_pose_; }
+  const bool &parameterization() const { return modDH_; }
+  const std::string parameterizationString() const {
+    if (!modDH_)
+      return "Denavit-Hartenberg";
+    else if (modDH_)
+      return "Modified Denavit-Hartenberg";
+  }
 
- private:
-  /// Calculate the homogenous tranformation and matrix for joint j with angle
+private:
+  /// Calculate the homogenous transformation and matrix for joint j with angle
   /// theta in the configuration space
   gtsam::Pose3 getJointTrans(size_t i, double theta) const {
     assert(i < dof());
     // DH transformation for each link, with theta matrix
-    return gtsam::Pose3(gtsam::Rot3::Rz(theta + theta_bias_(i)),
-                        gtsam::Point3(0, 0, 0)) *
-           link_trans_notheta_[i];
+    if (!modDH_) {
```

Comment:
Same deal here. If you check for `modDH_` then then you shouldn't need to check for `!modDH_`.